hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/fs/FileSystem.java

```diff
@@ -753,10 +755,47 @@ protected void checkPath(Path path) {
           return;
       }
     }
+
+    // The authority of the input path does not match with the
+    // authority of the client. Check if it matches with any
+    // of the available NNs in the system.
+    String nns = getConf().get("dfs.namenodes.rpc.addresses");
+    if (nns != null) {
+      uri = canonicalizeUri(uri);
+      String pathAuthority = uri.getAuthority();
+      StringTokenizer st = new StringTokenizer(nns, ",");
+      while (st.hasMoreTokens()) {
+        String nnAuthority = getNNAuthority(st.nextToken(), thisScheme);
+        if (nnAuthority != null && nnAuthority.equalsIgnoreCase(pathAuthority)) {
+          return;
```

Comment:
Is it good enough to return true? Will the calls to the NN happening after the checkPath call work eventhough it is not pointing to the same NN as the current dfs client?

Reply:
Yes, it will work.

Reply:
It is tested in the unit test